src/aiida_quantumespresso/workflows/functions/get_xspectra_structures.py

```diff
@@ -0,0 +1,332 @@
+
+        output_params['equivalent_sites_data'] = equivalency_dict
+
+        x_pos = []
+        y_pos = []
+        z_pos = []
+        for site in structure.sites:
+            x_pos.append(site.position[0])
+            y_pos.append(site.position[1])
+            z_pos.append(site.position[2])
+
+        # We assume that the Martyna-Tuckerman approach will be used in the core-hole SCF
+        # thus, at a minimum, the new box for the molecule needs to be 2x the extent of the
+        # molecule in x, y, and z. We therefore use the larger of either the MT distance or
+        # `supercell_min_parameter` for each cell parameter.
+        high_x = x_pos[(np.argmax(x_pos))]
+        high_y = y_pos[(np.argmax(y_pos))]
+        high_z = z_pos[(np.argmax(z_pos))]
+
+        low_x = x_pos[(np.argmin(x_pos))]
+        low_y = y_pos[(np.argmin(y_pos))]
+        low_z = z_pos[(np.argmin(z_pos))]
+
+        box_a = max((high_x - low_x) * 2, supercell_min_parameter)
+        box_b = max((high_y - low_y) * 2, supercell_min_parameter)
+        box_c = max((high_z - low_z) * 2, supercell_min_parameter)
+
+        centred_structure = StructureData()
+        centred_structure.set_cell(value=([box_a, 0., 0.], [0., box_b, 0.], [0., 0., box_c]))
+
+        for kind in structure.kinds:
+            centred_structure.append_kind(kind)
+
+        for site in structure.sites:
+            centred_structure.append_site(site)
+
+        # Reset the positions so that the centre of mass is at the centre of the new box
+        centred_structure.reset_sites_positions(centred_positions + (np.array(centred_structure.cell_lengths) / 2))
+
+        output_params['centred_structure_cell_matrix'] = centred_structure.cell
+        output_params['centred_structure_cell_lengths'] = centred_structure.cell_lengths
+        result['centred_structure'] = centred_structure
+
+        for value in equivalency_dict.values():
+            target_site = value['site_index']
+            marked_structure = StructureData()
+            centred_kinds = centred_structure.kinds
+
+            marked_structure.set_cell(centred_structure.cell)
+            for kind in centred_kinds:
+                marked_structure.append_kind(kind)
+
+            index_counter = 0
+            for site in centred_structure.sites:
+                if index_counter == target_site:
+                    absorbing_kind = Kind(name=abs_atom_marker, symbols=site.kind_name)
+                    absorbing_site = Site(kind_name=absorbing_kind.name, position=site.position)
+                    marked_structure.append_kind(absorbing_kind)
+                    marked_structure.append_site(absorbing_site)
+                else:
+                    new_site = Site(kind_name=site.kind_name, position=site.position)
+                    marked_structure.append_site(new_site)
+                index_counter += 1
+
+            result[f'site_{target_site}'] = marked_structure
```

Comment:
I suggest adding the symbol (kind) to the name too, so that one can easily know which element is calculated.

e.g. `site_2` to `site_2_Ti`

Reply:
I think it would be useful to add this information to the link label of the CalcFunction's outputs, for instance:

    Outputs                    PK  Type
    ----------------------  -----  -------------
    output_parameters       11298  Dict
    site_0_Li               11295  StructureData
    site_1_P                11296  StructureData
    site_2_F                11297  StructureData
    standardized_structure  11293  StructureData
    supercell               11294  StructureData

This would be useful for quick reference of the outputs of the CalcFunction at no extra effort. For the site labels in the `output_parameters` Dict node, the element information is kept in the dictionary itself, so I don't think adding it would be helpful - especially as the dictionary keys can return this information more reliably.

Reply:
Yes, I agree. Please add this.